src/routes/EResourceViewRoute.js

```diff
@@ -181,15 +192,18 @@ class EResourceViewRoute extends React.Component {
           entitlements: this.getRecords('entitlements'),
           packageContentsFilter: this.props.resources.packageContentsFilter,
           packageContents: this.getRecords('packageContents'),
+          packageContentsCount: get(this.props.resources, 'packageContents.other.totalRecords', 0),
         }}
         handlers={{
           ...handlers,
           onFilterPackageContents: this.handleFilterPackageContents,
+          onNeedMorePackageContents: this.handleNeedMorePackageContents,
           onClose: this.handleClose,
           onToggleTags: tagsEnabled ? this.handleToggleTags : undefined,
         }}
         helperApp={this.getHelperApp()}
         isLoading={this.isLoading()}
+        isPending={get(resources, 'packageContents.isPending', true)}
```

Comment:
I'm really not a fan of this.

1. It's a really generic prop name that you pass several levels deep. "_What's_ pending?"
1. It introduces another way of checking for whether a resource exists or not. Ultimately the purpose of this prop is so that we can [show a Spinner if a new filter is selected](https://github.com/folio-org/ui-agreements/pull/404/files#diff-118d7b474136e214559b4dd3b6fb33d9R139), right? Normally, when we need to a show a Spinner on initial load, we just check if the records are `undefined` (as opposed to an empty array). [Note the intentional lack of a default `[]` param for the `get()`.](https://github.com/folio-org/ui-erm/blob/bb06797ee945ecf13f2a866c4770ec8dddb33ebb/src/routes/AgreementViewRoute.js#L214)

So I'd say let's avoid passing any kind of `isPending` prop here. Instead, when `packageContents` is fetching the first page (upon initial load or after the filter is changed), pass its records as `undefined`. That may require tweaks to `this.getRecords`, and those tweaks may require https://github.com/folio-org/stripes-connect/pull/124.

Reply:
Another option that wouldn't require the stripes-connect could be to _not_ use `getRecords` for packageContents and write a custom `getPackageContentsRecords` for it. That function could check whether the records we already have are for an old filter because this component has access to a) the currently-selected package contents filter and b) the path used to fetch the current set of package contents (`resources.packageContents.url`). If that URL doesn't contain the current filter, we know the current set of records are stale and we should pass `data.packageContents` as `undefined` to our children.